series.go

```diff
@@ -31,6 +31,7 @@ type Metric struct {
 // Unit represents a unit definition that we might receive when query for timeseries data.
 // A metric is characterized by 2 units as: x per y
 // One or both could be missing
+type UnitPair *[2]*Unit
```

Comment:
The DD API isn't ideal for Go, so I'm not sure the best approach... but....

Instead of having an array of size 2 here, use a slice:
```go
type UnitPair []*Unit
```
And define a receiver method on Unit to manage it:
```go
func (u *Unit) SetUnit(unit, per_unit *Unit) {
  // Set u.Units appropriately
}
```

That's probably the best way to keep it understandable, given the DD API.

Reply:
Awesome, that was the pattern I was looking for. I'll try to get around to do it sometimes soon.

I'm still in the process of figuring out how to properly test things and hook up fixtures to properly test this PR.